### ISSUE TITLE
Remove dependency to annotation file for splicing pipeline

### ISSUE THREAD

**Issue by c-mertes** [gagneurlab/drop#170]
We have to rearrange the code so the full pipeline is not rerun each time some little bit changes in the sample annotation file.

Currently, the user-provided sample annotation is the first dependency for the aberrant splicing pipeline. This will make the pipeline run each time the sample annotation changes. 

Hence only make sure bam files of each sample provided within the annotation is the dependent input for aberrant splicing. 

Have a look [here](https://github.com/gagneurlab/drop/blob/master/drop/modules/aberrant-splicing-pipeline/Counting/00_define_datasets_from_anno.R#L12).

But we also have to make sure that if we change later the group association the pipeline gets re-executed if needed.

**Comment by mumichae** [comment]
So instead of using the sample annotation file as input, it would be better to evaluate the necessary sample runs via config and then feed that into the final target

**Comment by c-mertes** [comment]
Exactly. But we also have to make sure that the pipeline is rerun when dependencies change based on the grouping of samples. 

We could create an input file while checking the config and only update the file if the content changes. This could be done for the sample counting aka saving the count parameters per sample and sample annotation per group with the required columns for the specific pipeline parts. -> An example would be: `HPO terms` do not change how we run OUTRIDER/FRASER, but would change how we annotate the results.
